pkg/fetch/fetch.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2022 Chainguard, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fetch
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+)
+
+type Resource struct {
+	Host       string
+	Repository string
+	Path       string
+	Reference  string
+}
+
+func (r *Resource) URI() *url.URL {
+	return &url.URL{
+		Scheme: "https",
+		Host:   r.Host,
+		Path:   r.Repository,
+	}
+}
+
+func (r *Resource) String() string {
+	baseRef := strings.Join([]string{r.Host, r.Repository, r.Path}, "/")
+
+	if r.Reference != "" {
+		return fmt.Sprintf("%s@%s", baseRef, r.Reference)
+	}
+
+	return baseRef
+}
+
+func ParseRef(path string) (*Resource, error) {
+	paths, referenceName, _ := strings.Cut(path, "@")
+	pathElements := strings.Split(paths, string(os.PathSeparator))
```

Comment:
This should check whether there are enough pathElements to make this not panic later. Stuff like `include: github.com` or `include: github.com/foo@main` would be good tests too.